### PR TITLE
Adapt get_knownhosts to new facts structure

### PR DESCRIPTION
**Pull request by ppp0**

### CODE REVIEW

**Comment by ppp0** [comment]
@njam please review

**Comment by njam** [comment]
@kris-lab could you check?

File: modules/ssh/lib/puppet/parser/functions/get_knownhosts.rb [hunk 1]
```diff
@@ -4,10 +4,15 @@ module Puppet::Parser::Functions
     hostname = fqdn.split('.').shift
     aliases = []
     aliases.push hostname
-    lookupvar('::interfaces').split(',').each do |interface|
-      next if interface == 'lo'
-      ipaddress = lookupvar('::ipaddress_' + interface)
-      aliases.push ipaddress if ipaddress
+
+    networking_fact = lookupvar('networking')
```
Comment (kris-lab) [inline modules/ssh/lib/puppet/parser/functions/get_knownhosts.rb]: Should we maybe reference to tree-top like `::networking`?

File: modules/ssh/lib/puppet/parser/functions/get_knownhosts.rb [hunk 2]
```diff
@@ -4,10 +4,15 @@ module Puppet::Parser::Functions
...
+    interfaces = networking_fact['interfaces']
+
+    exclude_interfaces = [/^lo$/, /^vboxnet[\d]{1,2}$/]
+
+    interfaces.each do | interface_key,interface_value |
```
Comment (kris-lab) [inline modules/ssh/lib/puppet/parser/functions/get_knownhosts.rb]: Let's do formatting here?

**Comment by ppp0** [comment]
makes sense. @kris-lab please review

**Comment by kris-lab** [comment]
lgtm. let's wait for build.

**Comment by ppp0** [comment]
solves https://github.com/cargomedia/puppet-cargomedia/issues/1221

**Pull request merged**